radicle/src/ssh.rs

```diff
@@ -184,6 +241,146 @@ impl radicle_ssh::key::Private for SecretKey {
     }
 }
 
+#[derive(Debug, Error)]
+pub enum ExtendedSignatureError {
+    #[error(transparent)]
+    Base64Encoding(#[from] base64::DecodeError),
+    #[error("wrong preamble")]
+    MagicPreamble([u8; 6]),
+    #[error("missing armored footer")]
+    MissingFooter,
+    #[error("missing armored header")]
+    MissingHeader,
+    #[error(transparent)]
+    Encoding(#[from] encoding::Error),
+    #[error("public key encoding")]
+    PublicKeyEncoding,
+    #[error(transparent)]
+    PublicKeyError(#[from] PublicKeyError),
+    #[error("signature encoding")]
+    SignatureEncoding,
+    #[error(transparent)]
+    SignatureError(#[from] SignatureError),
+    #[error("unsupported version '{0}'")]
+    UnsupportedVersion(u32),
+}
+
+/// An SSH signature's decoded format.
+#[derive(Clone, Debug)]
+pub struct ExtendedSignature {
+    sig_version: u32,
+    publickey: crypto::PublicKey,
+    // unambigious interpretation domain to prevent cross-protocol attacks.
+    // Might be similar to 'git'.  Must be non-empty.
+    namespace: Vec<u8>,
+    // Should ignore
+    reserved: Vec<u8>,
+    // Hash used for signature
+    //  - sha256 or sha512
+    hash_algorithm: Vec<u8>,
+    signature: crypto::Signature,
+}
```

Comment:
The comments in this struct could be improved, or removed; right now they read like notes to self :smile:

Reply:
They were :(.  Fixed and added a link to SSH"s signature protocol document as a reference.